### PR TITLE
feat(deprecation): wrap deprecated functions

### PR DESCRIPTION
**Pull request by shsaskin**
Wraps deprecated functions with deprecatedOptions or util.deprecate.

Fixes Node-1430

### CODE REVIEW

File: lib/collection.js [hunk 1]
```diff
@@ -957,7 +958,7 @@ Collection.prototype.save = function(doc, options, callback) {
   }
 
   return executeOperation(this.s.topology, save, [this, doc, options, callback]);
-};
+}, 'colleciton.save is deprecated. Use insertOne, insertMany, updateOne, or updateMany instead.');
```
Comment (kvwalker) [inline lib/collection.js]: typo: 'colleciton'

File: lib/collection.js [hunk 2]
```diff
@@ -1290,12 +1301,12 @@ Collection.prototype.listIndexes = function(options) {
...
  * @return {Promise} returns Promise if no callback passed
  */
-Collection.prototype.ensureIndex = function(fieldOrSpec, options, callback) {
+Collection.prototype.ensureIndex = deprecate(function(fieldOrSpec, options, callback) {
   if (typeof options === 'function') (callback = options), (options = {});
   options = options || {};
 
   return executeOperation(this.s.topology, ensureIndex, [this, fieldOrSpec, options, callback]);
-};
+}, 'collection.ensureIndex is deprecated. Use createIndexes isntead.');
```
Comment (kvwalker) [inline lib/collection.js]: typo: 'isntead'

File: lib/db.js [hunk 3]
```diff
@@ -549,14 +554,14 @@ Db.prototype.listCollections = function(filter, options) {
...
-Db.prototype.eval = function(code, parameters, options, callback) {
+Db.prototype.eval = deprecate(function(code, parameters, options, callback) {
   const args = Array.prototype.slice.call(arguments, 1);
   callback = typeof args[args.length - 1] === 'function' ? args.pop() : undefined;
   parameters = args.length ? args.shift() : parameters;
   options = args.length ? args.shift() || {} : {};
 
   return executeOperation(this.s.topology, evaluate, [this, code, parameters, options, callback]);
-};
+}, 'Db.eval is deprecated on MongoDB 3.2 and forward');
```
Comment (kvwalker) [inline lib/db.js]: maybe phrase this as 'Db.eval is deprecated as of MongoDB 3.2'?

File: lib/db.js [hunk 4]
```diff
@@ -750,7 +755,7 @@ Db.prototype.ensureIndex = function(name, fieldOrSpec, options, callback) {
     options,
     callback
   ]);
-};
+}, 'Db.ensureIndex is deprecated since version 2.0');
```
Comment (kvwalker) [inline lib/db.js]: I think you should phrase this and the deprecation message for Db.eval similarly.

File: lib/collection.js [hunk 5]
```diff
@@ -1989,7 +2000,8 @@ Collection.prototype.group = function(
     options,
     callback
   ]);
-};
+},
+'MongoDB 3.6 or higher will no longer support the group command. We recommend rewriting using the aggregation framework.');
```
Comment (mbroadst) [inline lib/collection.js]: `MongoDB 3.6 or higher no longer supports`

**Pull request merged**